### PR TITLE
feat: add `uniques` function to array utilities

### PR DESCRIPTION
**Pull request by LordMidas**

### CODE REVIEW

**Review by TaroEld — APPROVED** [review]

File: msu/utils/array.nut [hunk 1]
```diff
@@ -104,4 +104,15 @@
 			}
 		}
 	}
+
+	// Returns a new array
+	function removeDuplicates( _array )
```
Comment (TaroEld) [inline msu/utils/array.nut:109]: Actually this should probably be called `uniqueValues` or `asSet` or something, as it  does not change the array in place.

Comment (LordMidas) [reply msu/utils/array.nut:109]: Yeah I see your point. `removeDuplicates` does sound like a function that should do its job in place. I don't like `asSet`. How about `filterUniques`?

Comment (TaroEld) [reply msu/utils/array.nut:109]: `filterUniques` sounds like it'll only give you duplicate values, lol.

Comment (LordMidas) [reply msu/utils/array.nut:109]: How about `uniques` or `getUniques`? I think `uniques` might be ok?

Comment (Enduriel) [reply msu/utils/array.nut:109]: I think `.uniques()` is fine actually, same idea as `.keys()` and `.values()` which I like

File: msu/utils/array.nut [hunk 2]
```diff
@@ -104,4 +104,15 @@
...
 		}
 	}
+
+	// Returns a new array
+	function uniques( _array )
+	{
+		local t = {};
+		foreach (a in _array)
+		{
+			t[a] <- true;
```
Comment (LordMidas) [inline msu/utils/array.nut:114]: This will fail if an element in the array is `null`. So we need to add handling for that. I'll work on that next.

Comment (LordMidas) [reply msu/utils/array.nut]: I have added handling for `null`. However it does not preserve the order in which `null` existed in the original array.

**Review by Enduriel — APPROVED** [review]

**Comment by TaroEld** [comment]
Needs the soluttion we discussed

File: msu/utils/array.nut (hunk 2 above)
Comment (LordMidas) [reply msu/utils/array.nut]: I've pushed a new commit based on the algorithm we decided on that preserves order and handles nulls.

**Review by TaroEld — APPROVED** [review]

**Pull request merged**